lib/internal/modules/esm/loader.js

```diff
@@ -423,13 +454,39 @@ class ESMLoader {
    * @returns {object}
    */
   async load(url, context = {}) {
-    const defaultLoader = this.#loaders[0];
+if (!this.isInternal) console.log('ESMLoader::load()', {
+  isInternal: this.isInternal,
+  url,
+  thisLoaders: this.#loaders,
+});
+    const loaders = this.#loaders.entries();
+    let {
+      0: loaderFilePath,
+      1: loader,
+    } = loaders.next().value;
+    let chainFinished = this.#loaders.size === 1;
```

Comment:
Using the `done` value seems less fragile, and probably more performant (no need to call `get Map.prototype.size`):
```suggestion
    let {
      value: {
        0: loaderFilePath,
        1: loader,
      },
      done: chainFinished,
    } = loaders.next();
```

Reply:
Oh! Yes, it does indeed. I forgot that will exist here 🙌